components/raftstore/src/store/worker/split_check.rs

```diff
@@ -234,20 +238,68 @@ where
             buckets.push(bucket_entry);
         }
 
-        if !buckets.is_empty() {
-            return self.router.send(
-                region.get_id(),
-                CasualMessage::RefreshRegionBuckets {
-                    region_epoch: region.get_region_epoch().clone(),
-                    buckets,
-                    bucket_ranges,
-                    cb: Callback::None,
-                },
-            );
-        }
+        self.on_buckets_created(&mut buckets, region, &ranges);
+        self.refresh_region_buckets(buckets, region, bucket_ranges);
         Ok(())
     }
 
+    fn on_buckets_created(
+        &self,
+        buckets: &mut Vec<Bucket>,
+        region: &Region,
+        bucket_ranges: &Vec<BucketRange>,
+    ) {
+        for (mut bucket, bucket_range) in &mut buckets.into_iter().zip(bucket_ranges) {
+            let adjusted_keys = std::mem::take(&mut bucket.keys)
+                .into_iter()
+                .enumerate()
+                .filter_map(|(i, key)| {
+                    let key = strip_timestamp_if_exists(key);
+                    let mut bucket_region = region.clone();
+                    bucket_region.set_start_key(bucket_range.0.clone());
+                    bucket_region.set_end_key(bucket_range.1.clone());
+                    if is_valid_split_key(&key, i, &bucket_region) {
+                        Some(key)
+                    } else {
+                        None
+                    }
+                })
+                .coalesce(|prev, curr| {
```

Comment:
This is almost the same as split check, why not reuse the code?

Reply:
As discussed, the parameter for is_valid_split_key is different--here it's per bucket range, not per region range.